ui/src/app/shared/workflow/sidebar/run-node/workflow.sidebar.run.node.component.ts

```diff
@@ -157,7 +154,8 @@ export class WorkflowSidebarRunNodeComponent implements OnDestroy, OnInit {
     }
 
     getCanBeRun(): boolean {
-        if (!this.workflow) {
+        if (!this.workflow || !this.workflow.environments || !this.workflow.environments
+            || !this.workflow.environments[this.node.context.environment_id]) {
```

Comment:
A workflow can have 0 environment

You can try to return if this.node is not in this.workflow